client/reducers/setDisabled.js

```diff
@@ -1,22 +1,17 @@
-import { SET_DISABLED, SET_DISABLED_LOADING } from '../actions'
+import { SET_JOINED, SET_LOADING } from '../actions'
 
 const initialState = {
   eventIDs: [],
-  loading: false,
-  success: false,
   failed: false
 }
 
 export function setDisabled (state = initialState, action) {
   switch (action.type) {
-    case SET_DISABLED:
-      return
+    case SET_JOINED:
+      return action.eventIds
 
-    case SET_DISABLED_LOADING:
-      return {
-        ...state,
-        loading: true
-      }
+    case SET_LOADING:
+      return action.loading
 
```

Comment:
Only one loading action instead of having an action for each 'state', passing in true or false when dispatch action is called.